includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -2253,19 +2253,23 @@ static function ( $refund ) use ( $refund_amount ) {
 			$currency = strtoupper( $refund['currency'] );
 			Tracker::track_admin( 'wcpay_edit_order_refund_success' );
 		} catch ( Exception $e ) {
+			if ( $e instanceof API_Exception && 'insufficient_balance_for_refund' === $e->get_error_code() ) {
+				// Handle insufficient_balance_for_refund error.
+				$this->order_service->handle_insufficient_balance_for_refund( $order, WC_Payments_Utils::prepare_amount( $amount, $order->get_currency() ) );
+			} else {
+				$note = sprintf(
+					/* translators: %1: the successfully charged amount, %2: error message */
+					__( 'A refund of %1$s failed to complete: %2$s', 'woocommerce-payments' ),
+					WC_Payments_Explicit_Price_Formatter::get_explicit_price( wc_price( $amount, [ 'currency' => $currency ] ), $order ),
+					$e->getMessage()
+				);
 
-			$note = sprintf(
-				/* translators: %1: the successfully charged amount, %2: error message */
-				__( 'A refund of %1$s failed to complete: %2$s', 'woocommerce-payments' ),
-				WC_Payments_Explicit_Price_Formatter::get_explicit_price( wc_price( $amount, [ 'currency' => $currency ] ), $order ),
-				$e->getMessage()
-			);
+				Logger::log( $note );
+				$order->add_order_note( $note );
+				$order->save();
```

Comment:
`$order->save()` is present in the `else` block but missing in the `if` block. Should we also save the order inside the `if` block, or maybe put it after if-else block??

Reply:
Good notice! I looked at this and it turned out that we even do not need `$order->save()` as `$order->add_order_note()` has already updated DB. I am removing `$order->save()` 9124c0c97e8f164aa330f21d9a9bf63ef3d06b5b

For the `if` block, `$order_service->handle_insufficient_balance_for_refund` has also run `add_order_note` to save the note to DB already so it does not need `$order->save()`.